packages/wdio-appium-service/src/launcher.ts

```diff
@@ -118,6 +118,13 @@ export default class AppiumLauncher implements Services.ServiceInstance {
     }
 
     async onPrepare() {
+        /**
+         * Throws an error if `this._options.args` is defined and is an array.
+         * @throws {Error} If `this._options.args` is an array.
+         */
+        if (this._options.args && Array.isArray(this._options.args)) {
```

Comment:
this will be sufficient
```suggestion
        if (Array.isArray(this._options.args)) {
```